java/ql/src/Violations of Best Practice/SpecialCharactersInLiterals/NonExplicitControlAndWhitespaceCharsInLiterals.md

```diff
@@ -0,0 +1,111 @@
+  | `\u0015`        | 21            | negative acknowledge      |
+  | `\u0016`        | 22            | synchronous idle          |
+  | `\u0017`        | 23            | end of transmission block |
+  | `\u0018`        | 24            | cancel                    |
+  | `\u0019`        | 25            | end of medium             |
+  | `\u001A`        | 26            | substitute                |
+  | `\u001B`        | 27            | escape                    |
+  | `\u001C`        | 28            | file separator            |
+  | `\u001D`        | 29            | group separator           |
+  | `\u001E`        | 30            | record separator          |
+  | `\u001F`        | 31            | unit separator            |
+  | `\u007F`        | 127           | delete                    |
+
+- Zero-width Unicode characters (e.g., zero-width space, zero-width joiner), such as:
+
+  | Escaped Unicode | Description               |
+  | --------------- | ------------------------- |
+  | `\u200B`        | zero-width space          |
+  | `\u200C`        | zero-width non-joiner     |
+  | `\u200D`        | zero-width joiner         |
+  | `\u2028`        | line separator            |
+  | `\u2029`        | paragraph separator       |
+  | `\u2060`        | word joiner               |
+  | `\uFEFF`        | zero-width no-break space |
+
+The following list outlines the _**explicit exclusions from query scope**_:
+
+- any number of simple space characters (`U+0020`, ASCII 32).
+- an escape character sequence (e.g., `\t`), or the Unicode equivalent (e.g., `\u0009`), for printable whitespace characters:
+
+  | Character Sequence | Escaped Unicode | ASCII Decimal | Description     |
+  | ------------------ | --------------- | ------------- | --------------- |
+  | `\t`               | \u0009          | 9             | horizontal tab  |
+  | `\n`               | \u000A          | 10            | line feed       |
+  | `\f`               | \u000C          | 12            | form feed       |
+  | `\r`               | \u000D          | 13            | carriage return |
+  |                    | \u0020          | 32            | space           |
+
+- character literals (i.e. single quotes) containing control characters.
+- literals defined within "likely" test methods, such as:
+  - JUnit test methods
+  - methods annotated with `@Test`
+  - methods of a class annotated with `@Test`
+  - methods with names containing "test"
+
+## References
+
+- [Unicode Control Characters](https://www.unicode.org/charts/PDF/U0000.pdf).
+- [Unicode C0 control codes](https://en.wikipedia.org/wiki/C0_and_C1_control_codes).
+- [Unicode characters with property "WSpace=yes" or "White_Space=yes"](https://en.wikipedia.org/wiki/Unicode_character_property#Whitespace).
+- [Java String Literals](https://docs.oracle.com/javase/tutorial/java/data/characters.html).
+- [Java Class Charset](https://docs.oracle.com/javase/8/docs/api///?java/nio/charset/Charset.html).
```

Comment:
It's a bit pedantic, but these don't follow the [style guide for references](https://github.com/github/codeql/blob/main/docs/query-help-style-guide.md#websites). Examples:
```
- Java API Specification: [Pattern.MULTILINE](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/regex/Pattern.html#MULTILINE)
- Wikipedia: [Template method pattern](https://en.wikipedia.org/wiki/Template_method_pattern).
```